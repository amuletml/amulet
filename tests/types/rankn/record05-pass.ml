external val (+) : int -> int -> int = ""

let apply
  : forall 'row 'a 'b.
    { 'row | apply : forall 'a 'b. ('a -> 'b) -> 'a -> 'b }
    -> ('a -> 'b) -> 'a -> 'b
  = fun r -> r.apply

let main =
  apply ({ apply = fun x -> x }) (fun x -> x + 1) 123
