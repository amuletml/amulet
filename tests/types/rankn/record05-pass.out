+ : int -> int -> int
apply : Spec{'row : type}. Spec{'a : type}. Spec{'b : type}. { 'row | apply : Spec{'a : type}. Spec{'b : type}. ('a -> 'b) -> 'a -> 'b } -> ('a -> 'b) -> 'a -> 'b
main : int
